src/test/java/org/folio/oaipmh/helpers/StorageHelperTest.java

```diff
@@ -93,6 +98,18 @@ void getItemId(String storageType) {
     assertThat(getStorageHelper(storageType).getRecordId(item), not(isEmptyOrNullString()));
   }
 
+  @Test
+  void shouldReturnLinkedToRecordInstanceId_whenGetRecordIdAndStorageIsSRS(){
+    JsonObject item = getJsonObjectFromFile(getDirPath(SOURCE_RECORD_STORAGE)+"/instance.json");
+    assertEquals(INSTANCE_ID, getStorageHelper(SOURCE_RECORD_STORAGE).getIdentifierId(item));
+  }
+
+  @Test
+  void shouldReturnEmptyString_whenGetRecordIdAndStorageIsSRSAndRecordHasNot999FieldWithLinkedInstanceId(){
```

Comment:
Are you sure that this is the correct behavior? The response from oai-pmh will have an empty id then, won't it?

Reply:
This is the righ behaviour. In case if record identifier are empty it means thar record doesn't have the metadata ("parsedRecord" part of json body) and such records will be filtered from response. So, there never won't be records with empty identifier.